kbchat/kbchat_test.go

```diff
@@ -103,82 +103,290 @@ func deleteWorkingDir(workingDir string) error {
+	require.NoError(t, err)
+	require.Greater(t, res.Result.MsgID, 0)
 
-func TestReactByChannel(t *testing.T) {}
+	// Read it to confirm it sent
+	messages, err := alice.GetTextMessages(teamChannel, false)
+	require.NoError(t, err)
+	sentMessage := messages[0]
+	require.Equal(t, sentMessage.Content.Type, "text")
+	require.Equal(t, sentMessage.Content.Text.Body, text)
+	require.Equal(t, sentMessage.MsgID, res.Result.MsgID)
+	testTeardown(alice, dir)
+}
+
+func TestSendAttachmentByTeam(t *testing.T) {
+	alice, _, dir, _, teamChannel := testSetup()
+	// Create a test file
+	fileName := "kb-attachment.txt"
+	location := path.Join(os.TempDir(), fileName)
+	data := []byte("My super cool attachment")
+	err := ioutil.WriteFile(location, data, 0644)
+	require.NoError(t, err)
+
+	// Send the message
+	title := "test SendAttachmentByTeam"
+	res, err := alice.SendAttachmentByTeam(teamChannel.Name, location, title, &teamChannel.TopicName)
+	require.NoError(t, err)
+	require.Greater(t, res.Result.MsgID, 0)
 
-func TestReactByConvID(t *testing.T) {}
+	// Types don't support attachments yet, so we can't read it
+	// Read it to confirm it sent
+	// messages, err := alice.GetTextMessages(teamChannel, false)
+	// require.NoError(t, err)
+	// sentMessage := messages[0]
+	// require.Equal(t, sentMessage.Content.Type, "attachment")
+	// require.Equal(t, sentMessage.Content.Attachment.Object.Title, title)
+	// require.Equal(t, sentMessage.MsgID, res.Result.MsgID)
+	testTeardown(alice, dir)
+}
+
+func TestReactByChannel(t *testing.T) {
+	alice, _, dir, oneOnOneChannel, _ := testSetup()
+	react := ":cool:"
+	// Get last message, we'll react to it
+	messages, err := alice.GetTextMessages(oneOnOneChannel, false)
+	require.NoError(t, err)
+	lastMessageID := messages[0].MsgID
+
+	// Send the react
+	res, err := alice.ReactByChannel(oneOnOneChannel, lastMessageID, react)
+	require.NoError(t, err)
+	require.Greater(t, res.Result.MsgID, 0)
+
+	// No great way to confirm reaction yet
+	testTeardown(alice, dir)
+}
+
+func TestReactByConvID(t *testing.T) {
+	alice, _, dir, oneOnOneChannel, _ := testSetup()
+	react := ":cool:"
+
+	// Get last message, we'll react to it
+	messages, err := alice.GetTextMessages(oneOnOneChannel, false)
+	require.NoError(t, err)
+	lastMessageID := messages[0].MsgID
+
+	// Retrieve conversation ID
+	messages, err = alice.GetTextMessages(oneOnOneChannel, false)
+	require.NoError(t, err)
+	convID := messages[0].ConversationID
+
+	// Send the react
+	res, err := alice.ReactByConvID(convID, lastMessageID, react)
+	require.NoError(t, err)
+	require.Greater(t, res.Result.MsgID, 0)
+	testTeardown(alice, dir)
+}
 
 func TestAdvertiseCommands(t *testing.T) {}
 
-func TestListChannels(t *testing.T) {}
+func TestListChannels(t *testing.T) {
+	alice, _, dir, _, teamChannel := testSetup()
+	channels, err := alice.ListChannels(teamChannel.Name)
+	require.NoError(t, err)
+	require.Greater(t, len(channels), 0)
```

Comment:
i can see how this might be open for debate, but i think i'd recommend doing something like:
```
require.True(t, len(channels)>0)
```
over the require.Greater. it seems less likely to result in someone accidentally putting the values in the wrong order.